rest/model/event.go

```diff
@@ -0,0 +1,65 @@
+package model
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/pkg/errors"
+)
+
+type APIEventLogEntry struct {
+	ID           *string        `bson:"_id" json:"-"`
+	ResourceType *string        `bson:"r_type,omitempty" json:"resource_type,omitempty"`
+	ProcessedAt  *time.Time     `bson:"processed_at" json:"processed_at"`
+	Timestamp    *time.Time     `bson:"ts" json:"timestamp"`
+	ResourceId   *string        `bson:"r_id" json:"resource_id"`
+	EventType    *string        `bson:"e_type" json:"event_type"`
+	Data         *TaskEventData `bson:"data" json:"data"`
+}
+
+type TaskEventData struct {
+	Execution int     `bson:"execution" json:"execution"`
+	HostId    *string `bson:"h_id,omitempty" json:"host_id,omitempty"`
+	UserId    *string `bson:"u_id,omitempty" json:"user_id,omitempty"`
+	Status    *string `bson:"s,omitempty" json:"status,omitempty"`
+	JiraIssue *string `bson:"jira,omitempty" json:"jira,omitempty"`
+
+	Timestamp *time.Time `bson:"ts,omitempty" json:"timestamp,omitempty"`
+	Priority  int64      `bson:"pri,omitempty" json:"priority,omitempty"`
+}
+
+func (el *APIEventLogEntry) BuildFromService(t interface{}) error {
+	switch v := t.(type) {
+	case *event.EventLogEntry:
+		d, ok := v.Data.(*event.TaskEventData)
+		if ok == false {
+			return errors.New(fmt.Sprintf("Incorrect type for data field when unmarshalling EventLogEntry"))
+		}
+		(*el) = APIEventLogEntry{
+			ID:           ToStringPtr(v.ID),
+			ResourceType: ToStringPtr(v.ResourceType),
+			ProcessedAt:  ToTimePtr(v.ProcessedAt),
+			Timestamp:    ToTimePtr(v.Timestamp),
+			ResourceId:   ToStringPtr(v.ResourceId),
+			EventType:    ToStringPtr(v.EventType),
+			Data: &TaskEventData{
```

Comment:
Extract this into a BuildFromService method on *TaskEventData, and do a switch case to determine the type to use (assuming that there will be support for other event types in the future)

Reply:
You don't ever call the function you added. Also you can change the parameter to the one you added to take in *event.TaskEventData - these function signatures are not actually part of any interface